src/framework/audio/internal/worker/externalaudiosource.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2022 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "externalaudiosource.h"
+using namespace mu;
+using namespace mu::audio;
+
+ExternalAudioSource::ExternalAudioSource(const TrackId trackId, const io::Device* playbackData)
```

Comment:
Why is the `const` necessary in `const io::Device*`? It looks like it isn't, and removing it makes the code simpler and better.

Reply:
Agree, I just tested it
```diff
diff --git a/src/framework/audio/internal/worker/externalaudiosource.cpp b/src/framework/audio/internal/worker/externalaudiosource.cpp
index d0d4067360..3bf057d2f4 100644
--- a/src/framework/audio/internal/worker/externalaudiosource.cpp
+++ b/src/framework/audio/internal/worker/externalaudiosource.cpp
@@ -23,12 +23,12 @@
 using namespace mu;
 using namespace mu::audio;
 
-ExternalAudioSource::ExternalAudioSource(const TrackId trackId, const io::Device* playbackData)
+ExternalAudioSource::ExternalAudioSource(const TrackId trackId, io::Device* playbackData)
     : m_trackId(trackId), m_playbackData(playbackData)
 {
     ONLY_AUDIO_WORKER_THREAD;
-    (const_cast<io::Device*>(playbackData))->open(mu::io::Device::ReadOnly);
-    b = (const_cast<io::Device*>(playbackData))->readAll();
+    playbackData->open(mu::io::Device::ReadOnly);
+    b = playbackData->readAll();
 
     drwav_init_memory(&m_wav, b.constData(), b.size(), nullptr);
     if (m_wav.fmt.bitsPerSample != 32) {
diff --git a/src/framework/audio/internal/worker/externalaudiosource.h b/src/framework/audio/internal/worker/externalaudiosource.h
index a2682909a3..4a6e2be373 100644
--- a/src/framework/audio/internal/worker/externalaudiosource.h
+++ b/src/framework/audio/internal/worker/externalaudiosource.h
@@ -29,7 +29,7 @@ class ExternalAudioSource : public ITrackAudioInput, public async::Asyncable
 {
 public:
     ExternalAudioSource();
-    explicit ExternalAudioSource(const TrackId trackId, const io::Device*);
+    explicit ExternalAudioSource(const TrackId trackId, io::Device *);
 
     bool isActive() const override;
     void setIsActive(const bool active) override;
``